miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleClassService.java

```diff
@@ -78,4 +85,25 @@ public void delete(Long sampleClassId) throws IOException {
     sampleClassDao.deleteSampleClass(sampleClass);
   }
 
+  @Override
+  public SampleClass inferStockFromAliquot(SampleClass sampleClass) {
+    SampleClass aliquotClass = sampleClassDao.getSampleClass(sampleClass.getId());
+    if (aliquotClass == null || !aliquotClass.getSampleCategory().equals(SampleAliquot.CATEGORY_NAME)) {
+      throw new IllegalArgumentException("Sample class " + sampleClass.getId() + " is not a valid aliquot class.");
+    }
+    SampleClass stockClass = null;
+    for (SampleValidRelationship relationship : sampleValidRelationshipDao.getSampleValidRelationship()) {
```

Comment:
Could make a query in SampleValidRelationship DAO for this instead of getting all and filtering in Java. Should perform a bit better, which may be relevant when bulk creating a large number (96) of samples